### PR TITLE
KAFKA-16107 [2/N]: Fix inverted args & more tests

### PR DESCRIPTION
**Pull request by lianetm**
Small fix for argument order in onPartitionsAssigned callback, and more tests for disabling/enabling partitions while callback runs

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ private CompletableFuture<Void> assignPartitions(
 
         // Invoke user call back.
         CompletableFuture<Void> result = invokeOnPartitionsAssignedCallback(addedPartitions);
-        result.whenComplete((error, callbackResult) -> {
+        result.whenComplete((callbackResult, error) -> {
```
Comment (dajac) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java]: nit: Should we rename `error` to `exception` or should we replace `callbackResult` by `__` as we don't use it?

**Review by dajac — COMMENTED** [review]
@lianetm Good catch. I left a small suggestion.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java (hunk 1 above)
Comment (lianetm) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java]: Sure, done

**Comment by lianetm** [comment]
Hey @dajac , comment addressed and rebased, thanks!

**Review by dajac — APPROVED** [review]
LGTM, thanks.

**Comment by dajac** [comment]
I triggered a new build as the last one timed out.

**Comment by dajac** [comment]
I was not able to get a clean build of this patch.

**Comment by lianetm** [comment]
Hey @dajac seems like we got a good [build](https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-15285/7/pipeline) now, and test failures are unrelated.

**Pull request merged**